### ISSUE TITLE
Import Failed

### ISSUE THREAD

**Issue by julyusito** [pfelk/pfelk#154]
Hello,

When I try to import the dashboard https://github.com/3ilson/pfelk/blob/master/Dashboard/v4.2%20(042020)%20Dashboard.ndjson , Get this error:

Failed to import 1 of 12 objects. Import failed

map [id=e3efbea0-865a-11ea-b7d8-d13c98fad18b] unsupported type

pfsense 2.4.5
Iam running the sebp/elk:latest container from dockerhub
I take the manual setup, based on the script
I get the messages ok
Ive created the index and everything else

Attached screens

Thanks for your help

![Captura de pantalla 2020-08-17 a la(s) 3 34 44 a  m](https://user-images.githubusercontent.com/69789966/90375741-fb79a600-e03a-11ea-8d8d-4e706c164132.png)
![Captura de pantalla 2020-08-17 a la(s) 3 30 16 a  m](https://user-images.githubusercontent.com/69789966/90375748-fddc0000-e03a-11ea-8f0a-441afe173427.png)
![Captura de pantalla 2020-08-17 a la(s) 3 29 49 a  m](https://user-images.githubusercontent.com/69789966/90375753-ff0d2d00-e03a-11ea-8b30-05861f7ec4a6.png)
![Captura de pantalla 2020-08-17 a la(s) 3 29 19 a  m](https://user-images.githubusercontent.com/69789966/90375754-ff0d2d00-e03a-11ea-8229-21727643b6b2.png)

**Comment by a3ilson** [comment]
@julyusito - you'll need to refresh your index...there should be >100 fields.  I believe you have a configuration issue within 01-inputs.conf based on the 27 fields and the provided screenshots.  Please provide line 11 and line 33 from within your 01-inputs.conf or utilize the error-data.sh (script) and provide the output on here, to better assist.

Ensure line 11 is pointing to your pfSenese host.  Also please confirm it is accessible (i.e. on the same network).
Ensure line 33 has the precedence "#pf#" removed and starts with "match...."

**Comment by julyusito** [comment]
Thanks for your response.

Ive attached the error log provided by the error-data.sh

[error.pfelk.log](https://github.com/3ilson/pfelk/files/5085828/error.pfelk.log)

![Captura de pantalla 2020-08-17 a la(s) 12 15 57 p  m](https://user-images.githubusercontent.com/69789966/90424571-f4c35100-e083-11ea-9d3e-e6909aa3182f.png)

**Comment by julyusito** [comment]
Hello @a3ilson 

Ive deploy another docker from zero with ubuntu, follow the automatic install, now I see the dashboard but the fields number is the same 27

This is my config from pfsense:
![Captura de pantalla 2020-08-17 a la(s) 2 29 22 p  m](https://user-images.githubusercontent.com/69789966/90436381-27764500-e096-11ea-9202-aba4146eb916.png)
![Captura de pantalla 2020-08-17 a la(s) 2 29 22 p  m](https://user-images.githubusercontent.com/69789966/90436389-2ba26280-e096-11ea-8a17-4a9f0eeba795.png)

And the dashboard show this
![Captura de pantalla 2020-08-17 a la(s) 2 31 19 p  m](https://user-images.githubusercontent.com/69789966/90436539-7b812980-e096-11ea-8fa7-ce22ac5d1ca5.png)

**Comment by a3ilson** [comment]
@julyusito - Thanks for the detailed information!  

Looks like we have a number of hurdles to clear with your modified setup.
(1) Since this is a Docker setup, you may need to remove lines 11-16 within 01-inputs.conf however, this does not appear to be preventing the logs from being received - skip this for now but make note.

(2) You have a second input via beats...logstash is running so doubtful that is an issue

(3) The 10-other.conf file was modified and the added filter duplicates the initial filter within 01-inputs.conf 

(4) You have an output at the end of 30-geoip.conf

Recommendation:
First, let's revert the configuration files to a working state (original).  Next add your modifications one at a time while restarting Logstash after each modification and ensuring an intact configuration.

**Comment by a3ilson** [comment]
@julyusito - #151 is similar problem set (e.g. ingesting different datasets).

**Comment by julyusito** [comment]
Hello,

Ive revert all config files, but the fields number is the same: 27
Maybe there is a problem parsing the logs?

**Comment by xchg-ax-ax** [comment]
I have the exact same problem, using the dockerized version. Only 27 fields and unable to import the dashboard.ndjson.
<img width="1674" alt="Screenshot 2020-08-19 at 09 27 55" src="https://user-images.githubusercontent.com/51474091/90605241-5ad3e380-e1fe-11ea-83c9-b3f1248ffe36.png">

**Comment by a3ilson** [comment]
@xchg-ax-ax - I responded to the issue opened on docker-pfelk.  Please update the conf files from there and if the problem still persist, open an issue within docker-pfelk.  

Thanks

**Comment by a3ilson** [comment]
@julyusito - please amend your 01-inputs.conf as follows:

```
# 01-inputs.conf
input {
  udp {
    port => 5140
    add_tag => [ "pf", "Ready" ]
    add_field => [ "[observer][type]", "firewall" ]
  }
}
filter {
  if "pf" in [tags] {
    grok {
      # OPNsense - Enable/Disable the line below based on firewall platform
      #OPN# match => { "message" => "<(?<[event][id]>.*)>%{SYSLOGTIMESTAMP:[event][created]} %{SYSLOGHOST:[observer][name]} %{DATA:labels}(?:\[%{POSINT:pf_pid}\])?: %{GREEDYDATA:pf_message}" }
      ########################################################################################################################################
      # pfSense - Enable/Disable the line below based on firewall platform
      match => { "message" => "<(?<[event][id]>.*)>%{SYSLOGTIMESTAMP:[event][created]} %{DATA:labels}(?:\[%{POSINT:[event][id]}\])?: %{GREEDYDATA:pf_message}" }
    }
    mutate {
      rename => { "[message]" => "[event][original]"}
      remove_tag => "Ready"
    }
  }
}
```

If the issues still exist after updating and restarting, please provide another error log file.

**Comment by julyusito** [comment]
Hello

Same result, but trigger this error in log: Unknown setting 'add_tag' for udp

[error.pfelk-2.log](https://github.com/3ilson/pfelk/files/5096929/error.pfelk-2.log)

**Comment by julyusito** [comment]
Found it!

My pfsense ip address was wrong!

For some reason, the pfsense source ip address was detected by elk docker like as virtualization station ip address; its a wrong behavior from my NAS, I'm escalating now to the fabric.

Now, I made 2 changes:
1) Change the 01-inputs.conf file pointing to the source address detected where the logs come from
2) Change the 05-firewall.conf file adjusting the interface names and alias from pfsense

Restart logstash
Reload pf-* index and voilá, I see now the 172 fields
Finally Import the dashboard and I have a beautiful glass from pfsense

[error.pfelk-3.log](https://github.com/3ilson/pfelk/files/5097220/error.pfelk-3.log)
![Captura de pantalla 2020-08-19 a la(s) 9 44 21 a  m](https://user-images.githubusercontent.com/69789966/90649722-a38c9c00-e200-11ea-83fd-d4e423d3e7ab.png)
